Bug #47452: Really restart slapd during USS

during the installation updates may be installed while still being in the
chroot environment. During that time "systemd" refused to (re-)start any
services, as "invoke-rc.d" calls "/sbin/runlevel", which returns:
> Running in chroot, ignoring request.

As "86_postinst_slapd_stop.patch" forcefully stops any old slapd, after that no
"slapd" is running any more. This hangs the installation as it retries to
authenticate to UMC, which then no longer works.

Insert code from "/usr/share/debhelper/autoscripts/postinst-init-restart" using
"service" instead of "invoke-rc.d", which does not honor "policy-rc.d".
diff --git debian/rules debian/rules
--- debian/rules
+++ debian/rules
@@ -161,7 +161,7 @@ endif
 		| xargs perl -pi -e 's#(\.TH \w+ 8)C#$$1#'
 
 override_dh_installinit:
-	dh_installinit -- "defaults 14 80"
+	dh_installinit --no-start -- "defaults 14 80"
 
 ifeq ($(filter stage1,$(DEB_BUILD_PROFILES)),)
 override_dh_installman:
diff --git debian/slapd.postinst debian/slapd.postinst
--- debian/slapd.postinst
+++ debian/slapd.postinst
@@ -324,6 +324,16 @@
 /etc/init.d/slapd stop
 
 #DEBHELPER#
+if [ "$1" = "configure" ] || [ "$1" = "abort-upgrade" ]; then
+	if [ -x "/etc/init.d/slapd" ]; then
+		if [ -n "$2" ]; then
+			_dh_action=restart
+		else
+			_dh_action=start
+		fi
+		service slapd $_dh_action || exit $?
+	fi
+fi
 
 exit 0
 
